lib/ecto/repo/supervisor.ex

```diff
@@ -110,11 +110,33 @@ defmodule Ecto.Repo.Supervisor do
             password: password,
             database: database,
             hostname: info.host,
-            port:     info.port]
+            port:     info.port] ++ parse_uri_query(info)
 
     for {k, v} <- opts, v, do: {k, if(is_binary(v), do: URI.decode(v), else: v)}
   end
 
+  defp parse_uri_query(%URI{query: nil}),
+    do: []
+  defp parse_uri_query(%URI{query: query} = url) do
+    query
+    |> URI.query_decoder()
+    |> Enum.reduce([], fn
+      {"ssl", "true"}, acc ->
```

Comment:
This is a subject of discussion, there are many ways to implement it. For example, maybe it's better to move all this stuff to a separate private function.